### DISCUSSION TITLE
Submitted Contracts

### DISCUSSION THREAD

**Discussion by thetroyharris** [duneanalytics/spellbook#3940] [New Sector Ideas]
Does anyone know if/how it is possible to reference the `blockchain.contracts_submitted` in a Spell. Both `ref(` and `source(` return an error.

**Comment by jeff-dude** [comment]
> Does anyone know if/how it is possible to reference the `blockchain.contracts_submitted` in a Spell. Both `ref(` and `source(` return an error.

can you link to PR where `source()` fails on this? it should work if it exists on the dune app. double check that you added the source to a source yaml file, where appropriate (likely a blockchain specific base_sources yaml file)

**Reply by hildobby** [reply]
@thetroyharris fairly certain that all you need here is to add the `contracts_submitted` tables to `base_sources` files (https://github.com/duneanalytics/spellbook/tree/main/models/base_sources)

**Reply by thetroyharris** [reply]
Thanks for the help guys. I'll submit a PR and close this discussion.
